packages/core/src/signals/src/signal.ts

```diff
@@ -10,6 +10,14 @@ import {createSignalFromFunction, defaultEquals, Signal, ValueEqualityFn} from '
 import {throwInvalidWriteToSignalError} from './errors';
 import {ReactiveNode} from './graph';
 
+/**
+ * If set, called after `WritableSignal`s are updated.
+ *
+ * This hook can be used to achieve various effects, such as running effects synchronously as part
+ * of setting a signal.
+ */
+let postSignalSetFn: (() => void)|null = null;
```

Comment:
optional / subjective: 
```suggestion
let afterSetCallback: (() => void)|null = null;
```